src/MethodProxies-Tests/MpConcurrencyHandler2.class.st

```diff
@@ -0,0 +1,13 @@
+Class {
+	#name : #MpConcurrencyHandler2,
+	#superclass : #MpHandler,
+	#category : #'MethodProxies-Tests'
+}
+
+{ #category : #evaluating }
+MpConcurrencyHandler2 >> beforeExecutionWithReceiver: anObject arguments: anArrayOfObjects [
+	anObject concurrencySemaphore signal.
+	anObject concurrencySemaphore wait.
+	anObject trigger2.
+	anObject concurrencySemaphore signal.
```

Comment:
Kind of the same, here, if the sequence is important, it's important to add comments here explaining why.
Concurrency here is too subtle and implicit. Comments can help to understand.